### PR TITLE
Fixed backport assignment issues.

### PR DESCRIPTION
**Pull request by russjones**
**Description**

Added ability to detect URLs for issues and deduplicate list of reviewers.

### CODE REVIEW

File: .github/workflows/robot/internal/bot/assign.go [hunk 1]
```diff
@@ -155,8 +155,26 @@ func (b *Bot) findOriginal(ctx context.Context, organization string, repository
...
+
+	return filtered
+}
+
 func isBackport(unsafeBase string) bool {
 	return strings.HasPrefix(unsafeBase, "branch/v")
 }
 
-var pattern = regexp.MustCompile(`#([0-9]+)`)
+var pattern = regexp.MustCompile(`(https:\/\/github\.com\/gravitational\/teleport\/pull\/|#)([0-9]+)`)
```
Comment (zmb3) [inline .github/workflows/robot/internal/bot/assign.go:180]: Why the change here?

Comment (russjones) [reply .github/workflows/robot/internal/bot/assign.go:180]: @espadolini Found an example where where the bot failed to detect a backport because the full PR URL was included instead of just the number: https://github.com/gravitational/teleport/pull/10944

Comment (espadolini) [reply .github/workflows/robot/internal/bot/assign.go:180]: Is `gravitational/teleport` the only repo where the bot runs?

Comment (espadolini) [reply .github/workflows/robot/internal/bot/assign.go:180]:
```suggestion
var pattern = regexp.MustCompile(`(?:https:\/\/github\.com\/gravitational\/teleport\/pull\/|#)([0-9]+)`)
```
and then we don't have to shift up the match numbers?

Comment (russjones) [reply .github/workflows/robot/internal/bot/assign.go:180]: Right now yes, but in the future we will probably pull it out into it's own action. I can clean this up but that will make it a runtime error.

File: .github/workflows/robot/internal/bot/assign.go [hunk 2]
```diff
@@ -124,15 +124,15 @@ func (b *Bot) findOriginal(ctx context.Context, organization string, repository
...
-	case len(title) == 2 && len(body) == 0:
-		original = title[1]
-	case len(title) == 2 && len(body) == 2:
-		if title[1] != body[1] {
+	case len(title) == 0 && len(body) == 3:
+		original = body[2]
+	case len(title) == 3 && len(body) == 0:
+		original = title[2]
+	case len(title) == 3 && len(body) == 3:
+		if title[2] != body[2] {
```
Comment (espadolini) [inline .github/workflows/robot/internal/bot/assign.go:132]:
```suggestion
	case len(title) == 0 && len(body) != 0:
		original = body[1]
	case len(title) != 0 && len(body) == 0:
		original = title[1]
	case len(title) != 0 && len(body) != 0:
		if title[1] != body[1] {
```

File: .github/workflows/robot/internal/bot/assign.go [hunk 3]
```diff
@@ -124,15 +124,15 @@ func (b *Bot) findOriginal(ctx context.Context, organization string, repository
...
 			return 0, trace.NotFound("different PRs referenced in title and body")
 		}
-		original = title[1]
+		original = title[2]
```
Comment (espadolini) [inline .github/workflows/robot/internal/bot/assign.go:135]:
```suggestion
		original = title[1]
```

File: .github/workflows/robot/internal/bot/assign.go (hunk 1 above)
Comment (espadolini) [inline .github/workflows/robot/internal/bot/assign.go:180]:
```suggestion
var pattern = regexp.MustCompile(`(?:https:\/\/github\.com\/gravitational\/teleport\/pull\/|#)([0-9]+)`)
```

**Review by espadolini — COMMENTED** [review]
Use a non-capturing group in the regexp:

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]
